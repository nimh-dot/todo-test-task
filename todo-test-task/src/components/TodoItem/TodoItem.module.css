.item {
    padding: 8px 4px;
    border-bottom: 2px solid rgb(237,237,237) ;
    display: flex;
    align-items: center;
    cursor: pointer;

}

.title {
    width: 80%;
    height: fit-content;
    text-wrap: wrap;
    overflow-wrap: break-word;
}

.checkbox {
    appearance: none;
    width: 25px;
    height: 25px;
    background: transparent;
    border: 1px solid rgb(107, 107, 107);
    border-radius: 50%;
}

.checkboxChecked {
    background: url('./tick.svg') no-repeat center center;
    background-size: cover;
}

.label {
    width: 100%;
    display: flex;
    align-items: center;
    gap: 10px;
    text-align: start;
    cursor: pointer;   
}

.labelChecked {
    text-decoration: line-through;
    opacity: .7;
}

.label:hover {
    border-radius: 3px;
    background-color: rgb(252, 245, 241);
}
