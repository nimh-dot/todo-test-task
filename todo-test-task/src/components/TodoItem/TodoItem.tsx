import { ITodo } from '../../types/data';
import styles from './TodoItem.module.css';
import cn from 'classnames';

interface ITodoItemProps {
    todo: ITodo
    toggleComplete: (id: number) => void
}

const TodoItem: React.FC<ITodoItemProps> = ({todo, toggleComplete}) => {
    const inputClass = cn({
        [styles.checkbox]: true,
        [styles.checkboxChecked]: todo.isComplete
    })

    const labelClass = cn({
        [styles.label]: true,
        [styles.labelChecked]: todo.isComplete
    })

    return (
        <li className={styles.item}>
            <label className={labelClass}>
                <input className={inputClass}
                    type="checkbox" 
                    checked={todo.isComplete} 
                    onChange={() => {toggleComplete(todo.id)}}
                />
                <span className={styles.title}>{todo.title}</span>
            </label>
        </li>
    )
}

export default TodoItem
