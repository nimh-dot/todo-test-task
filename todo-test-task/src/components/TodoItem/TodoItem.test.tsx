import { render, screen } from '@testing-library/react';
import TodoItem from './TodoItem';
import { ITodo } from '../../types/data';

const data: ITodo = {
    id: 3,
    title: 'предложить оффер',
    isComplete: false
}

describe('TodoItem', () => {
    it('renders the TodoItem component', () => {
      render(<TodoItem todo={data} toggleComplete={() => {}}/>)
      screen.debug()
    })
  
    it('list item and checkbox are present', () => {
      render(<TodoItem todo={data} toggleComplete={() => {}}/>)
      expect(screen.getByRole('checkbox')).toBeInTheDocument()
      expect(screen.getByRole('listitem')).toBeInTheDocument()
    })
  
    it('input field placeholder correct', () => {
      render(<TodoItem todo={data} toggleComplete={() => {}}/>)
      expect(screen.getByText(`${data.title}`)).toBeInTheDocument()
    })
  })