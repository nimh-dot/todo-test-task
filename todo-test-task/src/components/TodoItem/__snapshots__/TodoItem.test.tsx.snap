// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TodoItem > TodoItem snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="_item_00e4ad"
      >
        <label
          class="_label_00e4ad"
        >
          <input
            class="_checkbox_00e4ad"
            type="checkbox"
          />
          <span
            class="_title_00e4ad"
          >
            предложить оффер
          </span>
        </label>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="_item_00e4ad"
    >
      <label
        class="_label_00e4ad"
      >
        <input
          class="_checkbox_00e4ad"
          type="checkbox"
        />
        <span
          class="_title_00e4ad"
        >
          предложить оффер
        </span>
      </label>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
